### ISSUE TITLE
calculate KadabraBetweenness in directed graph 

### ISSUE THREAD

**Issue by hansonTang01** [networkit/networkit#1039]
when run the KadabraBetweenness for directed graph, error happens:
RuntimeError: Error, the diameter of directed graphs cannot be computed yet.

How can i fix it

**Comment by fabratu** [comment]
Hi,

unfortunately, our implementation of Kadabra-Betweenness does not support directed graphs yet. We should update the documentation accordingly.

**Comment by angriman** [comment]
The KadabraBetweenness algorithm supports directed graphs (see for example [here](https://github.com/networkit/networkit/blob/master/networkit/cpp/centrality/KadabraBetweenness.cpp#L564)). It cannot be used for directed graphs because it needs an estimation of the diameter of the graph, and Diameter does not support directed graphs yet.

**Comment by fabratu** [comment]
Yes, I could have been more precise here. That's what I meant with the "implementation of". We could also use a non-heuristic calculation of the diameter (maybe too time intensive) or another heuristic in case of directed graphs.

**Comment by angriman** [comment]
If I remember correctly, in our experiments we used this simple heuristic for directed graphs:
1. Pick a random node u, diam = 0.
2. Run a BFS from u and let v be the farthest node from u.
3. diam = max(diam, d(u, v)).
4. u = v, repeat 10 times from step 2.

It's cheap to run and the results were good enough. There might be better alternatives, though.

**Comment by fabratu** [comment]
Since this is not really an issue/bug, I will convert this thread to a discussion.
